src/OSInfo.cpp

```diff
@@ -23,6 +23,20 @@
 #include "Version.h"
 
 #if defined(Q_OS_WIN)
+// regString converts a wchar_t string of size to a
+// QString. If the string contains a NUL value, that
+// NUL value will terminate the string.
+static QString regString(wchar_t *string, int size) {
+	if (size <= 0) {
```

Comment:
And yes, I thought that == 0 might as well be there. By checking for <= 0, we're sure that we only pass postive integers to the functions, which is what we're sure they're coded to handle.